subsys/net/lib/nrf_cloud/src/nrf_cloud_agps.c

```diff
@@ -94,7 +104,21 @@ int nrf_cloud_agps_request(const struct nrf_modem_gnss_agps_data_frame *request)
 		types[type_count++] = NRF_CLOUD_AGPS_UTC_PARAMETERS;
 	}
 
-	if (request->sv_mask_ephe) {
+	ephem = request->sv_mask_ephe;
```

Comment:
In the AWS and Azure integrations in ATv2 we have our own encoding function for A-GPS requests, `cloud_codec_encode_agps_request`.

This functionality that seems to impact the A-GPS request format needs to be implemented in those cases as well. But, I don't think it makes sense to add it inside `cloud_codec_encode_agps_request`, since it is purely an encoding function.

A good place could be just before encoding in:

https://github.com/nrfconnect/sdk-nrf/blob/cfedfdfa08567b2252b511a4d1db15fbeba8152d/applications/asset_tracker_v2/src/modules/data_module.c#L673

Then you could pass in some flags/metadata into `cloud_codec_encode_agps_request` via `cloud_data_agps_request` that impacts the format of the message.

Reply:
The additional parameters are optional.  Requests without them are still supported.

That being said, yes, I can add additional entries to the `cloud_data_agps_request` structure.

Seems to me that `cloud_codec_encode_agps_request` is more of a wrapper around another function that actually does the encoding, `json_common_agps_request_data_add()`.  Who is going to make the changes there to encode it so your cloud sides are happy?

Reply:
I made the changes.  Let me know if you will be making the actual encoding changes.

Reply:
I can update the encoding after this is merged, @coderbyheart needs to be in the loop. I've made a ticket:

https://projecttools.nordicsemi.no/jira/browse/IRIS-2637